datapipelines/src/test/java/io/customer/datapipelines/support/core/UnitTest.kt

```diff
@@ -48,21 +49,29 @@ abstract class UnitTest {
         SDKComponent.registerAnalyticsFactory { moduleConfig ->
             return@registerAnalyticsFactory createTestAnalyticsInstance(moduleConfig)
         }
-        // Register Android SDK component with mocked dependencies as we are using real context in tests
+        // Register AndroidSDKComponent with mocked dependencies as we are not using real context in tests
+        mockAndroidSDKComponent()
+    }
+
+    protected open fun mockAndroidSDKComponent() {
+        // Setup AndroidSDKComponent by mocking dependencies that depends on Android context
+        // Prefer relaxed mocks to avoid unnecessary setup for methods that are not used in the test
         val androidSDKComponent = SDKComponent.registerAndroidSDKComponent(
-            context = mock(),
+            context = mockk(relaxed = true),
             client = Client.Android(Version.version)
         )
         // Mock global preference store to avoid reading/writing to shared preferences
-        globalPreferenceStore = mock()
-        androidSDKComponent.overrideDependency(GlobalPreferenceStore::class.java, globalPreferenceStore)
+        globalPreferenceStore = mockk<GlobalPreferenceStore>(relaxUnitFun = true).also { instance ->
+            androidSDKComponent.overrideDependency(GlobalPreferenceStore::class.java, instance)
+        }
+        every { globalPreferenceStore.getDeviceToken() } returns null
```

Comment:
why do we need this? it wasn't before?

Reply:
What we were previously doing with `Mockito` was actually using relaxed mocks. Here, I used `relaxUnitFun = true` so for functions expected to return any non-unit value, we have to explicitly mention them. So we have more control over them and can have clarity on test results with the expected behavior of these functions.